operators/endpointmetrics/controllers/observabilityendpoint/ocp_monitoring_config.go

```diff
@@ -458,185 +409,182 @@ func createOrUpdateClusterMonitoringConfig(
-		return err
+	} else {
+		foundClusterMonitoringConfiguration.PrometheusK8sConfig = newPmK8sConfig
 	}
-	found.Data[clusterMonitoringConfigDataKey] = string(updatedclusterMonitoringConfigurationYAMLBytes)
-	err = client.Update(ctx, found)
+
+	updatedClusterMonitoringConfigurationYAMLBytes, err := yaml.Marshal(foundClusterMonitoringConfiguration)
 	if err != nil {
-		log.Error(err, "failed to update configmap", "name", clusterMonitoringConfigName)
-		return err
+		log.Error(err, "failed to marshal the cluster monitoring config to yaml")
 	}
-	log.Info("configmap updated", "name", clusterMonitoringConfigName)
-	return unset(ctx, client, namespace)
-}
 
-// unset config reverted flag after successfully updating cluster-monitoring-config
-func unset(ctx context.Context, client client.Client, ns string) error {
-	// if reverted before, reset so we can revert again
-	revertedAlready, err := isRevertedAlready(ctx, client, ns)
-	if err == nil && revertedAlready {
-		err = unsetConfigReverted(ctx, client, ns)
-	}
-	return err
+	found.Data[clusterMonitoringConfigDataKey] = string(updatedClusterMonitoringConfigurationYAMLBytes)
+	return updateClusterMonitoringConfigAndUnset(ctx, client, found, namespace)
 }
 
 // revertClusterMonitoringConfig reverts the configmap cluster-monitoring-config and relevant resources
 // (observability-alertmanager-accessor and hub-alertmanager-router-ca) for the openshift cluster monitoring stack.
 func revertClusterMonitoringConfig(ctx context.Context, client client.Client) error {
 	log.Info("revertClusterMonitoringConfig called")
 
-	// try to retrieve the current configmap in the cluster
 	found := &corev1.ConfigMap{}
-	err := client.Get(ctx, types.NamespacedName{Name: clusterMonitoringConfigName,
-		Namespace: promNamespace}, found)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			log.Info("configmap not found, no need action", "name", clusterMonitoringConfigName)
-			return nil
-		} else {
-			log.Error(err, "failed to check configmap", "name", clusterMonitoringConfigName)
-			return err
-		}
-	}
-
-	// do not touch the configmap if are not already a manager
-	touched := false
-	log.Info("checking field.Manager")
-	for _, field := range found.GetManagedFields() {
-		log.Info("filed.Manager", "manager", field.Manager)
-		if field.Manager == endpointMonitoringOperatorMgr {
-			touched = true
-			break
-		}
+	err := client.Get(ctx, types.NamespacedName{Name: clusterMonitoringConfigName, Namespace: promNamespace}, found)
+	if err != nil && !errors.IsNotFound(err) {
+		log.Error(err, "failed to check configmap", "name", clusterMonitoringConfigName)
+		return err
 	}
 
-	if !touched {
-		log.Info(
-			"endpoint-monitoring-operator is not a manager of configmap, no action needed",
-			"name",
-			clusterMonitoringConfigName,
-		)
+	log.Info("configmap exists, check if it needs revert", "name", clusterMonitoringConfigName)
```

Comment:
The CM does not necessarily exists at this point. Maybe update the message or change the logic?